src/components/TodoHeader.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { ErrorMessage } from '../types/ErrorMessage';
+
+type Props = {
+  error: ErrorMessage;
+  isToogleAll: boolean | null;
+  isInputDisablet: boolean;
+  isDeletedTodos: number;
+  onAddTodo: (title: string) => Promise<void>;
+  onToggleAll: () => void;
+  setErrorMessage: (error: ErrorMessage) => void;
+};
+
+export const TodoHeader: React.FC<Props> = props => {
+  const {
+    error,
+    isToogleAll,
+    isInputDisablet,
+    isDeletedTodos,
+    onAddTodo,
+    onToggleAll,
+    setErrorMessage,
+  } = props;
+
+  const [title, setTitle] = useState('');
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [inputRef, isInputDisablet, isDeletedTodos]);
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (error !== ErrorMessage.Default) {
+      setErrorMessage(ErrorMessage.Default);
+    }
+
+    if (!title.trim()) {
+      setErrorMessage(ErrorMessage.EmptyTitle);
+
+      return;
+    }
+
+    return onAddTodo(title.trim())
+      .then(() => setTitle(''))
+      .catch(() => {});
```

Comment:
You should handle errors. If you have done it previously you don't need a catch at all